osf/models/node.py

```diff
@@ -461,10 +466,23 @@ def has_published_preprint(self):
 
     @property
     def preprint_url(self):
+        if self.linked_preprint:
+            return self.linked_preprint.url
+
+    @property
+    def has_moderated_preprint(self):
+        if self.is_preprint:
+            return self.linked_preprint.provider.reviews_workflow
+
+    @property
+    def linked_preprint(self):
         if self.is_preprint:
             try:
                 # if multiple preprints per project are supported on the front end this needs to change.
-                return self.preprints.filter(is_published=True)[0].url
+                if self.has_published_preprint:
+                    return self.preprints.filter(is_published=True)[0]
```

Comment:
`has_published_preprint` already does this exact query (with `exists()`), it seems wasteful to do it again.